larrecodnn/NuGraph/NuGraphInferenceSonicTriton_module.cc

```diff
@@ -125,432 +55,123 @@ class NuGraphInferenceSonicTriton : public art::EDProducer {
-      if (planes[p] == "u") {
-        size_t n_rows = x_semantic_u_data.size() / n_cols;
-        s = torch::from_blob(x_semantic_u_data.data(),
-                             {static_cast<int64_t>(n_rows), static_cast<int64_t>(n_cols)},
-                             options);
-      }
-      else if (planes[p] == "v") {
-        size_t n_rows = x_semantic_v_data.size() / n_cols;
-        s = torch::from_blob(x_semantic_v_data.data(),
-                             {static_cast<int64_t>(n_rows), static_cast<int64_t>(n_cols)},
-                             options);
-      }
-      else if (planes[p] == "y") {
-        size_t n_rows = x_semantic_y_data.size() / n_cols;
-        s = torch::from_blob(x_semantic_y_data.data(),
-                             {static_cast<int64_t>(n_rows), static_cast<int64_t>(n_cols)},
-                             options);
-      }
-      else {
-        std::cout << "Error!!" << std::endl;
-      }
-
-      for (int i = 0; i < s.sizes()[0]; ++i) {
-        size_t idx = idsmap[p][i];
-        std::array<float, 5> input({s[i][0].item<float>(),
-                                    s[i][1].item<float>(),
-                                    s[i][2].item<float>(),
-                                    s[i][3].item<float>(),
-                                    s[i][4].item<float>()});
-        softmax(input);
-        FeatureVector<5> semt = FeatureVector<5>(input);
-        (*semtcol)[idx] = semt;
-      }
-    }
-  }
-  if (filterDecoder) {
-    for (size_t p = 0; p < planes.size(); p++) {
-      torch::Tensor f;
-      torch::TensorOptions options = torch::TensorOptions().dtype(torch::kFloat32);
-      if (planes[p] == "u") {
-        int64_t num_elements = x_filter_u_data.size();
-        f = torch::from_blob(x_filter_u_data.data(), {num_elements}, options);
-      }
-      else if (planes[p] == "v") {
-        int64_t num_elements = x_filter_v_data.size();
-        f = torch::from_blob(x_filter_v_data.data(), {num_elements}, options);
-      }
-      else if (planes[p] == "y") {
-        int64_t num_elements = x_filter_y_data.size();
-        f = torch::from_blob(x_filter_y_data.data(), {num_elements}, options);
-      }
-      else {
-        std::cout << "error!" << std::endl;
-      }
-
-      for (int i = 0; i < f.numel(); ++i) {
-        size_t idx = idsmap[p][i];
-        std::array<float, 1> input({f[i].item<float>()});
-        (*filtcol)[idx] = FeatureVector<1>(input);
-      }
-    }
+  //
+  // Get pointers to the result returned and write to the event
+  //
+  vector<NuGraphOutput> infer_output;
+  for (const auto& pair : infer_result) {
+    const auto& prob = pair.second.fromServer<float>();
+    size_t n_elements = std::distance(prob[0].begin(), prob[0].end());
+    std::vector<float> out_data;
+    out_data.reserve(n_elements);
+    out_data.insert(out_data.end(), prob[0].begin(), prob[0].end());
+    infer_output.push_back(NuGraphOutput(pair.first, out_data));
   }
```

Comment:
```suggestion
  for (auto const& [name, data] : infer_result) {
    auto const& prob = data.fromServer<float>();
    std::vector<float> out_data(prob[0].begin(), prob[1].end());
    infer_output.emplace_back(name, std::move(out_data));
  }
```

Reply:
note that this suggestion was implemented but with `prob[0]` instead of `prob[1]`, which I assumed is a typo

Reply:
Good catch, @cerati.